docs/content/preview/tutorials/google/google-vertex-ai.md

```diff
@@ -0,0 +1,163 @@
+   ```
+## Getting Started with Google Vertex AI
+To start using Google Vertex AI in the application:
+1. Create a project in Google Cloud.
+1. Enable the Vertex AI API.
+1. Install the [gcloud command-line utility](https://cloud.google.com/sdk/docs/install).
+1. Log-in to Google Cloud via the CLI to run the application locally.
+  ```shell
+  gcloud auth application-default login
+  ```
+1. Update the Google Vertex AI environment variables in `{project_dir}/backend/.env`.
+
+## Generate Embeddings for Airbnb Listing Descriptions
+
+Airbnb properties come with detailed descriptions of the accommodations, location, and various other features of the listing. By transforming the text in the `description` column of our database into a vectorized representation, we can use `pgvector` to execute a similarity search based on user prompts.
+
+Execute the `embeddingsGenerator.js` script to generate embeddings in Google Vertex AI for each property, and store them in the `description_embedding` column in the database. This process can take over 10 minutes to complete.
+
+```shell
+   node {project_directory}/backend/vertex/embeddingsGenerator.js
+   ....
+   Processing rows starting from 34746755
+   Processed 7551 rows
+   Processing rows starting from 35291912
+   Finished generating embeddings for 7551 rows
+```
+
+
+## Running the Application
+
+1. Start the Node.js backend.
+   ```
+   cd {project_dir}/backend
+   npm start
+   ```
+1. Start the React UI.
+   ```
+   npm run dev
+   ```
+1. Access the application UI at [http://localhost:5173](http://localhost:5173).
+
+
+## Testing the Application
+
+
+Test the application with relevant prompts. For instance:
+
+   *We're traveling to San Francisco from February 21st through 28th. We need a place to stay with parking available.*
+
+   *I'm looking for an apartment near the Golden Gate Bridge with a nice view of the Bay.*
+
+   *Full house with ocean views for a family of 6.*
+
+   *Room for 1 in downtown SF, walking distance to Moscone Center.*
+
+These prompts will first be sent to Vertex AI to be converted to embeddings. Next, the returned embeddings will be used to search for similar properties, stored in YugabyteDB.
+
+```javascript
+const dbRes = await pool.query(
+     "SELECT name, description, price, 1 - (description_embedding <=> $1) as similarity " +
+       "FROM airbnb_listing WHERE 1 - (description_embedding <=> $1) > 0.7 ORDER BY similarity DESC LIMIT 5",
+     [embeddings]
+   );
+```
+![YugaLodgings Application Search Results](/images/tutorials/google/google-vertex-ai/yugalodgings-search-results.png "YugaLodgings Application Search Results")
+
+## Wrapping Up
+
+Building generative AI applications using YugabyteDB is easily achieved with the help of PostgreSQL's pgvector extension and Google Vertex AI. If you're interested in more tutorials on the topic of generative AI, check out:
+
+[Link to Azure OpenAI Tutorial]
```

Comment:
@ddhodge I wasn't sure how to link most effectively to other tutorials within our docs. Do I just do a relative path from this tutorial? How is this managed when the docs structure changes and things are moved around?

Reply:
We use relative links. I can take care of that.